h2o-automl/src/test/java/ai/h2o/automl/leaderboard/LeaderboardTest.java

```diff
@@ -0,0 +1,167 @@
+
+import ai.h2o.automl.AutoML;
+import ai.h2o.automl.events.EventLog;
+import hex.Model;
+import hex.tree.gbm.GBM;
+import hex.tree.gbm.GBMModel;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.Keyed;
+import water.Lockable;
+import water.fvec.Frame;
+import water.util.Log;
+import water.util.TwoDimTable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class LeaderboardTest extends water.TestUtil {
+
+  // TODO: plenty of tests to write for this class
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  private static Key<AutoML> dummy = Key.make();
+
+  @Test
+  public void test_toTwoDimTable_with_empty_models_and_without_sort_metric() {
+    Leaderboard lb = null;
+    EventLog eventLog = EventLog.getOrMake(dummy);
+    try {
+      lb = Leaderboard.getOrMake("dummy_lb_no_sort_metric", eventLog, new Frame(), null);
+
+      TwoDimTable table = lb.toTwoDimTable();
+      assertNotNull("empty leaderboard should also produce a TwoDimTable", table);
+      assertEquals("no models in this leaderboard", table.getTableDescription());
+    } finally {
+      if (lb != null) lb.remove();
+      eventLog.remove();
+    }
+  }
+
+  @Test
+  public void test_toTwoDimTable_with_empty_models_and_with_sort_metric() {
+    Leaderboard lb = null;
+    EventLog eventLog = EventLog.getOrMake(dummy);
+    try {
+      lb = Leaderboard.getOrMake("dummy_lb_logloss_sort_metric", eventLog, new Frame(), "logloss");
+
+      TwoDimTable table = lb.toTwoDimTable();
+      assertNotNull("empty leaderboard should also produce a TwoDimTable", table);
+      assertEquals("no models in this leaderboard", table.getTableDescription());
+    } finally {
+      if (lb != null) lb.remove();
+      eventLog.remove();
+    }
+  }
+
+
+  @Test
+  public void test_rank_tsv() {
+    Leaderboard lb = null;
+    EventLog eventLog = EventLog.getOrMake(dummy);
+    GBMModel model = null;
+    Frame fr  = null;
+    try {
+      fr = parse_test_file("./smalldata/logreg/prostate_train.csv");
+      GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+      parms._train = fr._key;
+      parms._nfolds = 2;
+      parms._seed = 1;
+      parms._response_column = "CAPSULE";
+      GBM job = new GBM(parms);
+      model = job.trainModel().get();
+      
+      lb = Leaderboard.getOrMake("dummy_rank_tsv", eventLog, null, "mae");
+      lb.addModel(model._key);
+      Log.info(lb.rankTsv());
+      assertEquals("Error\n[0.3448260574357465, 0.19959320678410908, 0.44675855535636816, 0.19959320678410908, 0.31468498072970547]\n", lb.rankTsv());
+    } finally {
+      if (lb != null){
+        lb.remove();
+      }
+      eventLog.remove();
+      if (model != null) {
+        model.deleteCrossValidationModels();
+        model.delete();
+      }
+      if( fr != null) {
+        fr.delete();
+      }
+    }
+  }
+
+  @Test
```

Comment:
only this test is new.
The rest of the test class was moved when moving `Leaderboard` to its dedicated package.

Reply:
@deil87 to answer your question about Java tests: this big test should cover all the Leaderboard extensions feature.